spec/yaml/twilio_accounts_v1.yaml

```diff
@@ -3,92 +3,80 @@ components:
+          $ref: '#/components/schemas/url'
       type: object
     accounts.v1.credential:
       properties: {}
       type: object
     accounts.v1.credential.credential_aws:
       properties:
         account_sid:
-          maxLength: 34
-          minLength: 34
-          pattern: ^AC[0-9a-fA-F]{32}$
-          type: string
+          $ref: '#/components/schemas/sid_AC'
         date_created:
-          format: date-time
-          type: string
+          $ref: '#/components/schemas/date_time_iso8601'
         date_updated:
-          format: date-time
-          type: string
+          $ref: '#/components/schemas/date_time_iso8601'
         friendly_name:
-          type: string
+          $ref: '#/components/schemas/string'
         sid:
-          maxLength: 34
-          minLength: 34
-          pattern: ^CR[0-9a-fA-F]{32}$
-          type: string
+          $ref: '#/components/schemas/sid_CR'
         url:
-          format: uri
-          type: string
+          $ref: '#/components/schemas/url'
       type: object
     accounts.v1.credential.credential_public_key:
       properties:
         account_sid:
-          maxLength: 34
-          minLength: 34
-          pattern: ^AC[0-9a-fA-F]{32}$
-          type: string
+          $ref: '#/components/schemas/sid_AC'
         date_created:
-          format: date-time
-          type: string
+          $ref: '#/components/schemas/date_time_iso8601'
         date_updated:
-          format: date-time
-          type: string
+          $ref: '#/components/schemas/date_time_iso8601'
         friendly_name:
-          type: string
+          $ref: '#/components/schemas/string'
         sid:
-          maxLength: 34
-          minLength: 34
-          pattern: ^CR[0-9a-fA-F]{32}$
-          type: string
+          $ref: '#/components/schemas/sid_CR'
         url:
-          format: uri
-          type: string
+          $ref: '#/components/schemas/url'
       type: object
     accounts.v1.secondary_auth_token:
       properties:
         account_sid:
-          maxLength: 34
-          minLength: 34
-          pattern: ^AC[0-9a-fA-F]{32}$
-          type: string
+          $ref: '#/components/schemas/sid_AC'
         date_created:
-          format: date-time
-          type: string
+          $ref: '#/components/schemas/date_time_iso8601'
         date_updated:
-          format: date-time
-          type: string
+          $ref: '#/components/schemas/date_time_iso8601'
         secondary_auth_token:
-          type: string
+          $ref: '#/components/schemas/string'
         url:
-          format: uri
-          type: string
+          $ref: '#/components/schemas/url'
       type: object
+    date_time_iso8601:
+      format: date-time
+      type: string
+    sid_AC:
+      maxLength: 34
+      minLength: 34
+      pattern: ^AC[0-9a-fA-F]{32}$
+      type: string
+    sid_CR:
+      maxLength: 34
+      minLength: 34
+      pattern: ^CR[0-9a-fA-F]{32}$
+      type: string
+    string:
```

Comment:
I don't think it has a directly negative effect. The reason `string` is defined as a top level schema type is because `string` is a [type used in api-definitions and needs to map to some sort of schema in the oai spec](https://code.hq.twilio.com/twilio/yps/blob/main/yps/json_schema.py#L205). I think the complex logic to parse out api-definition type names that match primitive oai spec types isn't worth the effort it would take to implement, given that the interpretation of the spec in both cases is identical.

Reply:
Hmm makes sense. Looks odd that's all. I guess if you don't see any build errors in twilio-go, maybe the generator also has a way to handle such namings.

Reply:
The generator resolves down to the referenced schema, which is just the standard string type